lib/service/src/main/java/io/atlasmap/service/AtlasService.java

```diff
@@ -113,6 +119,42 @@ public void onUpdate(AtlasLibraryLoader loader) {
         });
     }
 
+    /**
+     * The user has specified an ADM file on inception.  Clear current mappings and process
+     * the specified ADM catalog file.
+     *
+     * @param atlasmapAdmPath - user specified ADM catalog file
+     * @throws AtlasException
+     */
+    private void initializeADMCatalog(String atlasmapAdmPath) throws AtlasException {
+        LOG.debug("Initializing with user specified catalog ADM path {}", atlasmapAdmPath);
+        try {
+            java.nio.file.Path admPath = Paths.get(atlasmapAdmPath);
+            java.nio.file.Path mappingFolderPath = Paths.get(mappingFolder);
+            File atlasDir = new File(mappingFolder);
+            if (!atlasDir.exists()) {
+                atlasDir.mkdirs();
+            }
+            atlasDir = new File(libFolder);
+            if (!atlasDir.exists()) {
+                atlasDir.mkdirs();
+            }
+
+            resetMappings();
+
+            AtlasUtil.copyFile(admPath.toAbsolutePath(),
+                Paths.get(mappingFolderPath.toAbsolutePath().toString() + File.separator + atlasmapCatalogName));
+
+            extractCompressedCatalog();
+
+        } catch (Exception e) {
+            LOG.error("Error processing the AtlasMap catalog file " + atlasmapAdmPath + "\n" +
+                e.getMessage(), e);
+            throw new WebApplicationException(e.getMessage(), e, Status.INTERNAL_SERVER_ERROR);
+        }
+        System.setProperty(ATLASMAP_ADM_PATH, "");
```

Comment:
why are you setting the value to empty value? it means that after launch and first use the property provided by the user is overridden? Why not keeping it to know on which transformation it was initially launched?

Reply:
The user can import a totally different ADM file with the import icon.  Keeping ATLASMAP_ADM_PATH pointing to the original file causes it to be reread even of the user imported a different one from the UI.

Reply:
ok